### PR TITLE
Add v15 CHANGELOG breaking notes about container images

### PR DESCRIPTION
**Pull request by wadells**
[Rendered](https://github.com/gravitational/teleport/blob/walt/quay-turndown/CHANGELOG.md#container-images)

This PR adds user facing info about turning off our quay.io image hosting, in addition to other breaking v15 OCI image changes.  I've been plotting the quay.io deprecation schedule, and I think the v15 release notes are as good of a place as any to make the announcement.

@camscale asked me to add notes about distroless and the Teleport Operator too.

I'm worried this is a bit wordy, but I didn't find a shorter way to convey all the needed info.

The (tenative) quay.io brownout/retirement schedule is:
* 2024-05-01 - 3 hours during a busy time of day
* 2024-05-15 - a day
* 2024-06-03 - a week
* 2024-07-03 - make quay repos private
* 2025-07-XX - Delete all quay data and terminate our accounts/relationship

Contributes to https://github.com/gravitational/SecOps/issues/289

### CODE REVIEW

**Comment by wadells** [comment]
`no-changelog` I guess?  Or is this 100% changelog.  I'm not sure.  What is the right thing to do here @fheinecke?

File: CHANGELOG.md [hunk 1]
```diff
@@ -12,6 +12,58 @@ rely on parsing the output from multiple nodes should pass the `--log-dir` flag
 to `tsh ssh`, which will create a directory where the separated output of each node
 will be written.
 
+#### Container Images
+
+Teleport 15 contains several breaking changes to improve the default
+security and usability of container images.
+
+##### "Fat" container images are discontinued
```
Comment (camscale) [inline CHANGELOG.md:20]: s/Fat/Heavy/ - in Mac land, fat means multi-architecture and we are going harder on that.

File: CHANGELOG.md [hunk 2]
```diff
@@ -12,6 +12,58 @@ rely on parsing the output from multiple nodes should pass the `--log-dir` flag
...
+
+For users who need a shell in a Teleport container, a "debug" image is
+available which contains BusyBox, including a shell and many CLI tools. Find the debug images at:
+
+* https://gallery.ecr.aws/gravitational/teleport-distroless-debug
+* https://gallery.ecr.aws/gravitational/teleport-ent-distroless-debug
+
+Do not run debug container images in production environments.
+
+Fat container images will continue to be published for Telport 13 and 14
```
Comment (camscale) [inline CHANGELOG.md:39]: s/Fat/Heavy/

File: CHANGELOG.md [hunk 3]
```diff
@@ -12,6 +12,58 @@ rely on parsing the output from multiple nodes should pass the `--log-dir` flag
...
+throughout the remainder of these release's lifecycle.
+
+##### Multi-architecture Teleport Operator Images
+
+Teleport Operator container images will no longer be published with architecture
+suffixes in their tags (for example: `14.2.1-amd64` and `14.2.1-arm`). Instead,
+only a single tag will be published with multi-platform support (e.g. 15.0.0)
+If you use Teleport Operator images with a version suffix, remove the suffix
+and your client should automatically pull the platform appropriate image.
+Individual architectures may be pulled with the `docker pull --platform <arch>`.
```
Comment (camscale) [inline CHANGELOG.md:49]: s/the// - or add "command" at the end.

File: CHANGELOG.md [hunk 4]
```diff
@@ -12,6 +12,58 @@ rely on parsing the output from multiple nodes should pass the `--log-dir` flag
...
+longer supported and no new images will be published to quay.io.
+
+For supported versions of Teleport, please find the distroless container images
+in [Teleport's public ECR registry](https://gallery.ecr.aws/gravitational).
+
+Following brownouts in May and June, Teleport will disable pulls from
+quay.io repos on Wednesday, 2024-07-03.
+
+Users who wish to continue to use Teleport v12 and earlier beyond July 2024
+will need to download the unsupported quay.io images they depend on and mirror
```
Comment (camscale) [inline CHANGELOG.md:64]: Or they could just switch to the ECR images which will still be there.

**Review by camscale — APPROVED** [review]

**Comment by camscale** [comment]
> What is the right thing to do here

I guess you just create an empty commit and put the entire changelog in the PR on the "Changelog:" line :rofl: 

The no-changelog label is correct - it means we won't look for a line to add to the changelog from this PR when making a release.

File: CHANGELOG.md (hunk 1 above)
Comment (wadells) [reply CHANGELOG.md:20]: Yeah -- I didn't love "Fat" either, but "distrofull" seemed worse.  Will update to heavy.

File: CHANGELOG.md (hunk 4 above)
Comment (camscale) [reply CHANGELOG.md:64]: I see. I was only thinking about v12 users and the time it would take them to transition to a supported version. I mean, who uses an unsupported version of a security product? I guess it's a convenience product too for some and the security is an added benefit.

Should we copy all the old images into ECR? Maintain them for the historical record when in 50 years when they do a documentary on how Teleport took over the world, they can go back to the early versions and still run them?

File: CHANGELOG.md (hunk 1 above)
Comment (wadells) [reply CHANGELOG.md]: fixed in https://github.com/gravitational/teleport/pull/35452/commits/8407fc98aa1005736ecef01fe60d1e47cf140dea

File: CHANGELOG.md (hunk 2 above)
Comment (wadells) [reply CHANGELOG.md]: fixed in https://github.com/gravitational/teleport/pull/35452/commits/8407fc98aa1005736ecef01fe60d1e47cf140dea

**Review by ptgott — APPROVED** [review]
